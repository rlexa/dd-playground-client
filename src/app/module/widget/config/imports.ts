import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
