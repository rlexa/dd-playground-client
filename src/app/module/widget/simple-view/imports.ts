import { CommonModule } from '@angular/common';
import { MatButtonModule, MatListModule } from '@angular/material';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RipupperPipeModule } from 'app/module/pipe/ripupper';
import { StartuppercasePipeModule } from 'app/module/pipe/startuppercase';
import { FlexboxModule } from 'dd-flexbox';

export const imports = [CommonModule, MatButtonModule, MatListModule, FlexboxModule, IconPipeModule, RipupperPipeModule, StartuppercasePipeModule];
