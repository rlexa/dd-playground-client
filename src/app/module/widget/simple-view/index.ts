export * from './simple-view.module';
