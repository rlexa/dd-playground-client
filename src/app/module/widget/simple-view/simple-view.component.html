<div *ngIf="data$ | async as model" class="position-relative">
  <div ddFlexbox="hys" [style.maxWidth]="'100%'" class="scroll-x-auto">
    <button mat-button *ngIf="isExpandable" class="inline-button" (click)="isExpanded = !isExpanded">
      <i class="material-icons rot-90">{{ (isExpanded ? 'collapse' : 'expand') | icon }}</i>
    </button>
    <mat-list *ngIf="!isExpandable || isExpanded" class="mat-list-auto-height" [attr.dense]="isDense ? true : null">
      <table *ngIf="keys$ | async as keys">
        <ng-container *ngIf="types$ | async as types">
          <ng-container *ngIf="clickables$ | async as clickables">
            <tr *ngFor="let key of keys; let iindex = index;">
              <td [style.verticalAlign]="'top'">
                <mat-list-item>
                  <span class="no-break-text">{{ key | startuppercase | ripupper }}</span>
                </mat-list-item>
              </td>
              <!-- <td class="contenthv" [style.verticalAlign]="'top'">{{ types[iindex] }}</td> -->
              <td>
                <mat-list-item [@TRIGGER_WobbleX]="['recursive'].includes(types[iindex]) ? null : model[key]">
                  <span class="break-text" [ngClass]="{'value-numeric': types[iindex] === 'number', 'size-data-label': isWidthConstrained}">
                    <ng-container [ngSwitch]="types[iindex]">
                      <a *ngSwitchCase="'url'" [href]="model[key]">{{ model[key] }}</a>
                      <span *ngSwitchCase="'timestamp'" [ngClass]="{'clickable': clickables[key], 'color-accent': clickables[key]}"
                        (click)="clickables[key] ? onClicked(key, model[key]) : {}">
                        {{ model[key] | date:FORMAT_DATE_TIMESTAMP}}
                      </span>
                      <ng-container *ngSwitchCase="'json'"> {{ model[key] | json }} </ng-container>
                      <ng-container *ngSwitchCase="'void'"> {{ '-' }} </ng-container>
                      <app-simple-view *ngSwitchCase="'recursive'" [data]="model[key]" [isWidthConstrained]="isWidthConstrained"
                        [isExpandable]="true" [isExpanded]="false" [isClickable]="isClickable" (clicked)="onClicked($event.key, $event.value)"></app-simple-view>
                      <span *ngSwitchDefault [ngClass]="{'clickable': clickables[key], 'color-accent': clickables[key]}" (click)="clickables[key] ? onClicked(key, model[key]) : {}">
                        {{ model[key] }}
                      </span>
                    </ng-container>
                  </span>
                </mat-list-item>
              </td>
            </tr>
          </ng-container>
        </ng-container>
      </table>
    </mat-list>
  </div>
</div>