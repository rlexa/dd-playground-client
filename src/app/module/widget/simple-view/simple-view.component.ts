import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import { TRIGGER_WobbleX } from 'app/animations';
import { FORMAT_DATE_TIMESTAMP } from 'app/presets';
import { isNumeric, isWeb, WithDataProperty } from 'app/util';
import { map } from 'rxjs/operators';

type CellType = 'url' | 'number' | 'timestamp' | 'json' | 'recursive' | 'string' | 'void';

@Component({
  selector: 'app-simple-view',
  templateUrl: './simple-view.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
  animations: [TRIGGER_WobbleX],
})
export class SimpleViewComponent extends WithDataProperty<Object>  {
  readonly FORMAT_DATE_TIMESTAMP = FORMAT_DATE_TIMESTAMP;

  readonly keys$ = this.data$.pipe(map(_ => Object.keys(_ || {})));
  readonly types$ = this.keys$.pipe(map(_ => _.map(key => this.data$.value[key]).map(this.guessType)));
  readonly clickables$ = this.data$.pipe(map(_ => Object.entries(_ || {})
    .reduce((acc, [key, value]) => ({ ...acc, [key]: !this.isClickable ? false : this.isClickable(key, value) }), <{ [key: string]: boolean }>{})));

  @Input() isWidthConstrained = true;
  @Input() isClickable: (key: string, val: any) => boolean = null;
  @Input() isDense = false;
  @Input() isExpanded = true;
  @Input() isExpandable = false;
  @Input() subheader = <string>null;

  @Output() clicked = new EventEmitter<{ key: string, value: any }>();

  onClicked = (key: string, value: any) => this.clicked.emit({ key, value });

  private guessType = (val: any): CellType => val === null || val === undefined ? 'void' :
    typeof val === 'object' ? val instanceof Date ? 'timestamp' : 'recursive' :
      typeof val === 'number' ? 'number' :
        typeof val === 'string' ? isWeb(val) ? 'url' : isNumeric(val) ? 'number' : 'string' : 'json';

}
