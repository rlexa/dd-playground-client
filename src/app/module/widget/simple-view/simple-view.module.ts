import { NgModule } from '@angular/core';
import { imports } from './imports';
import { SimpleViewComponent } from './simple-view.component';

@NgModule({ imports, exports: [SimpleViewComponent], declarations: [SimpleViewComponent] })
export class SimpleViewModule { }
