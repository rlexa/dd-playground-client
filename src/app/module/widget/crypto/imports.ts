import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
} from '@angular/material';
import { RouterModule } from '@angular/router';
import { CryptoApiModule } from 'app/module/service/crypto-api';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  FormsModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
  FlexboxModule,
  CryptoApiModule,
  SimpleViewModule,
];
