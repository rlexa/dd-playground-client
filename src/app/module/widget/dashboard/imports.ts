import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
