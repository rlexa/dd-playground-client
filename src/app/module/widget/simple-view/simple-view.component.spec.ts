import {async, TestBed} from '@angular/core/testing';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';
import {imports} from './imports';
import {SimpleViewComponent} from './simple-view.component';

describe('SimpleViewComponent', () => {
  beforeEach(async(() => {
    TestBed.configureCompiler({preserveWhitespaces: false} as any)
      .configureTestingModule({
        imports: [NoopAnimationsModule, ...imports],
        declarations: [SimpleViewComponent],
        providers: [],
      })
      .compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(SimpleViewComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
