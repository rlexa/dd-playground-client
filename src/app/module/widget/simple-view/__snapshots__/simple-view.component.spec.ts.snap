// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleViewComponent renders 1`] = `
<mock-render>
  <app-simple-view
    class="ng-tns-c3480983330-0"
    ng-reflect-is-expanded="true"
    ng-reflect-is-width-constrained="true"
  >
  </app-simple-view>
</mock-render>
`;

exports[`SimpleViewComponent with data renders 1`] = `
<mock-render>
  <app-simple-view
    class="ng-tns-c3480983330-0"
    ng-reflect-data="[object Object]"
    ng-reflect-is-expanded="true"
    ng-reflect-is-width-constrained="true"
  >
    <div
      class="position-relative ng-tns-c3480983330-0 ng-star-inserted"
    >
      <div
        class="list-h scroll-x-auto ng-tns-c3480983330-0"
        style="max-width: 100%;"
      >
        <mat-list
          class="mat-list-auto-height ng-tns-c3480983330-0 ng-star-inserted"
        >
          <table
            class="ng-tns-c3480983330-0 ng-star-inserted"
          >
            <tr
              class="ng-tns-c3480983330-0 ng-star-inserted"
            >
              <td
                class="ng-tns-c3480983330-0"
                style="vertical-align: top;"
              >
                <mat-list-item
                  class="ng-tns-c3480983330-0"
                >
                  <span
                    class="no-break-text"
                  >
                     undefined 
                  </span>
                </mat-list-item>
              </td>
              <td
                class="ng-tns-c3480983330-0"
              >
                <mat-list-item
                  class="ng-tns-c3480983330-0 ng-trigger ng-trigger-TRIGGER_WobbleX"
                >
                  <span
                    class="break-text size-data-label"
                    ng-reflect-ng-class="[object Object]"
                  >
                    <span
                      class="ng-tns-c3480983330-0 ng-star-inserted"
                      ng-reflect-ng-class="[object Object]"
                      style=""
                    >
                       value 
                    </span>
                  </span>
                </mat-list-item>
              </td>
            </tr>
          </table>
        </mat-list>
      </div>
    </div>
  </app-simple-view>
</mock-render>
`;
